sdk/identity/azure-identity/src/azure_cli_credential.cpp

```diff
@@ -120,8 +113,12 @@ AzureCliCredential::AzureCliCredential(const Core::Credentials::TokenCredentialO
 std::string AzureCliCredential::GetAzCommand(std::string const& scopes, std::string const& tenantId)
     const
 {
-  ThrowIfNotSafeCmdLineInput(scopes, "Scopes");
-  ThrowIfNotSafeCmdLineInput(m_tenantId, "TenantID");
+  // The OAuth 2.0 RFC (https://datatracker.ietf.org/doc/html/rfc6749#section-3.3) allows space as
+  // well for a list of scopes, but that isn't currently required.
+  ThrowIfNotSafeCmdLineInput(
+      scopes, ".-:/_", "Scopes"); // Characters allowed in the scope are [0-9a-zA-Z-.:/_].
+  ThrowIfNotSafeCmdLineInput(
+      m_tenantId, ".-", "TenantID"); // Characters allowed in the tenant id are [0-9a-zA-Z-.].
```

Comment:
nit: I find these "Characters allowed in the ... are ..." a to excessive - you write in the comment what the code clearly does in the very same line - not it is only obvious, but also the comment may get out of sync and become confusing.